### PR TITLE
ENH: Number of simulations on simualte var

### PR DESCRIPTION
**Pull request by EC-AI**
- [ ] closes #xxxx
- [ ] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by EC-AI** [comment]
Added code to simulate several times in one call, taking advantage of matrix operations so its much faster than calling the function n times.

File: statsmodels/tsa/vector_ar/util.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ def acf_to_acorr(acf):
     return acf / np.sqrt(np.outer(diag, diag))
 
 
-def varsim(coefs, intercept, sig_u, steps=100, initial_values=None, seed=None):
+def varsim(coefs, intercept, sig_u, steps=100, initial_values=None, seed=None, n_sim=1):
```
Comment (bashtage) [inline statsmodels/tsa/vector_ar/util.py:194]: Can we use `nsimulations` which is the same as `SARIMAX`

use `nsimulations=None`.

`None` means 1 simulation with shape (steps, neqs).
Any integer value means (nsimulations, steps, neqs)

File: statsmodels/tsa/vector_ar/util.py [hunk 2]
```diff
@@ -225,42 +225,47 @@ def varsim(coefs, intercept, sig_u, steps=100, initial_values=None, seed=None):
...
         If seed is not None, then it will be used with for the random
         variables generated by numpy.random.
+    n_sim : int
+        Number of simulations to perform.
 
     Returns
     -------
     endog_simulated : nd_array
-        Endog of the simulated VAR process
+        Endog of the simulated VAR process. Shape will be (n_sim, steps, neqs)
```
Comment (bashtage) [inline statsmodels/tsa/vector_ar/util.py:234]: Please update after other change

File: statsmodels/tsa/vector_ar/var_model.py [hunk 3]
```diff
@@ -923,7 +923,7 @@ def is_stable(self, verbose=False):
         """
         return is_stable(self.coefs, verbose=verbose)
 
-    def simulate_var(self, steps=None, offset=None, seed=None, initial_values=None):
+    def simulate_var(self, steps=None, offset=None, seed=None, initial_values=None, n_sim=1):
```
Comment (bashtage) [inline statsmodels/tsa/vector_ar/var_model.py:926]: Same change here, where `None` is the default value.

File: statsmodels/tsa/vector_ar/util.py [hunk 4]
```diff
@@ -225,42 +225,47 @@ def varsim(coefs, intercept, sig_u, steps=100, initial_values=None, seed=None):
...
+        or (steps, neqs) if `n_sim` = 1.
     """
     rs = np.random.RandomState(seed=seed)
     rmvnorm = rs.multivariate_normal
     p, k, k = coefs.shape
     if sig_u is None:
         sig_u = np.eye(k)
-    ugen = rmvnorm(np.zeros(len(sig_u)), sig_u, steps)
-    result = np.zeros((steps, k))
+    ugen = rmvnorm(np.zeros(len(sig_u)), sig_u, steps*n_sim).reshape(n_sim, steps, k)
```
Comment (bashtage) [inline statsmodels/tsa/vector_ar/util.py:242]: <import int_like at top>

```python
nsimulations= int_like(nsimulations, "n_sim", optional=True)
if isinstance(nsimulations, int) and n_sim <= 0:
    raise ValueError("nsimulationsmust be a positive integer if provided")
```

**Review by bashtage — CHANGES_REQUESTED** [review]
Biggest is to rename to `nsimulations`.  Some other to make it more robust.

Thanks.

**Comment by EC-AI** [comment]
The tests I included are wrong. Wait.

**Review by bashtage — APPROVED** [review]
LGTM. Thanks.

**Comment by bashtage** [comment]
Looks good.  Thanks.

**Pull request merged**